cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -419,5 +419,56 @@ CUDA_HOST_DEVICE_CALLABLE constexpr decltype(auto) type_dispatcher(cudf::data_ty
   }
 }
 
+namespace detail {
+template <typename T1>
+struct dispatch_second_type {
+#pragma nv_exec_check_disable
+  template <typename T2, typename F, typename... Ts>
+  CUDA_HOST_DEVICE_CALLABLE decltype(auto) operator()(F&& f, Ts&&... args) const
+  {
+    return f.template operator()<T1, T2>(std::forward<Ts>(args)...);
+  }
+};
+
+template <template <cudf::type_id> typename IdTypeMap>
+struct dispatch_first_type {
+#pragma nv_exec_check_disable
+  template <typename T1, typename F, typename... Ts>
+  CUDA_HOST_DEVICE_CALLABLE decltype(auto) operator()(cudf::data_type type2,
+                                                      F&& f,
+                                                      Ts&&... args) const
+  {
+    return type_dispatcher<IdTypeMap>(
+      type2, detail::dispatch_second_type<T1>{}, std::forward<F>(f), std::forward<Ts>(args)...);
+  }
+};
+}  // namespace detail
+
+/**
+ * @brief Dispatches two type template parameters to a callable.
+ *
+ * This function expects a callable `f` with an `operator()` template accepting
+ * two typename template parameters `T1` and `T2`.
+ *
+ * @param type1 The `data_type` used to dispatch a type for the first template
+ * parameter of the callable `F`
+ * @param type2 The `data_type` used to dispatch a type for the second template
+ * parameter of the callable `F`
+ * @param args Parameter pack forwarded to the `operator()` invocation `F`.
+ */
+#pragma nv_exec_check_disable
+template <template <cudf::type_id> typename IdTypeMap = id_to_type_impl, typename F, typename... Ts>
+CUDA_HOST_DEVICE_CALLABLE constexpr decltype(auto) type_double_dispatcher(cudf::data_type type1,
```

Comment:
I think `double_type_dispatcher` would be more appropriate.

Reply:
Resolved in d58245ae942c1c34882f30b0d4c3dba904031c21.